services/nostr-band/nostr-band-followers.service.js

```diff
@@ -0,0 +1,57 @@
+import Joi from 'joi'
+import { metric } from '../text-formatters.js'
+import { BaseJsonService } from '../index.js'
+
+const npubSchema = Joi.object({
+  followers_pubkey_count: Joi.number().required(),
+}).required()
+
+const mainSchema = Joi.object({
+  stats: Joi.object().pattern(Joi.string(), npubSchema).required(),
```

Comment:
If we're going to access `Object.keys(stats)[0]` we need to make sure the stats object has at least one key when we validate. This schema will accept `{ "stats": {} }` as a valid input and then produce undesired output. We can ensure the object has at least one key by doing `stats: Joi.object().min(1).pattern(...` here. Tbh, I'd actually be inclined to make this `.min(1).max(1)` as based on the code, it seems like we're really expecting _exactly_ one key and two would be unexpected?

Reply:
done: https://github.com/badges/shields/pull/9870/commits/4d28b105572e9d5f716c4898e152d81a31e9ac80